test/unit/server.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+var chai = require('chai');
+chai.use(require('chai-as-promised'));
+var assert = chai.assert;
+var sinon = require('sinon');
+
+var Promise = require('bluebird');
+var Worker = require('../../lib/worker');
+var hermes = require('runnable-hermes');
+var noop = require('101/noop');
+var ponos = require('../../');
+
+var tasks = {
+  'test-queue-01': worker,
+  'test-queue-02': worker
+};
+function resolvePromise () { return Promise.resolve(); }
+function worker (job, done) {}; // eslint-disable-line no-unused-vars
+
+describe('Ponos Server', function () {
+  var server;
+  before(function () { sinon.stub(Worker, 'create'); });
+  beforeEach(function () {
+    server = ponos.server({ queues: Object.keys(tasks) });
+    sinon.stub(server.hermes, 'connectAsync', resolvePromise);
+    sinon.stub(server.hermes, 'subscribeAsync', resolvePromise);
+  });
+  afterEach(function () {
+    server.hermes.connectAsync.restore();
+    server.hermes.subscribeAsync.restore();
+  });
+  after(function () { Worker.create.restore(); });
+
+  describe('Constructor', function () {
+    beforeEach(function () {
+      sinon.stub(hermes, 'hermesSingletonFactory').returns(noop);
+    });
+    afterEach(function () { hermes.hermesSingletonFactory.restore(); });
+
+    it('should take a Hermes client', function () {
+      var s = ponos.server({ hermes: noop });
+      assert.ok(s);
+    });
+    it('should require a list of queues', function () {
+      assert.throws(function () {
+        ponos.server();
+      }, /missing.+queues/);
+    });
+    it('should require a string list of queues', function () {
+      assert.throws(function () {
+        ponos.server({ queues: [{}] });
+      }, /queues.+string/);
+    });
+  });
+
+  describe('setTask', function () {
+    it('should accept tasks, and not subscribe', function () {
+      var queue = Object.keys(tasks)[0];
+      return assert.isFulfilled(server.setTask(queue, worker))
+        .then(function () {
+          assert.notOk(server.hermes.subscribeAsync.calledOnce);
+          assert.isFunction(server._tasks[queue]);
+        });
+    });
+  });
+
+  describe('setAllTasks', function () {
+    it('should set multiple tasks', function () {
+      sinon.spy(server, 'setTask');
+      var numTasks = Object.keys(tasks).length;
+      return assert.isFulfilled(server.setAllTasks(tasks))
+        .then(function () {
+          assert.equal(server.setTask.callCount, numTasks);
+          assert.equal(Object.keys(server._tasks).length, numTasks);
+        });
+    });
+  });
+
+  describe('start', function () {
+    describe('missing tasks', function () {
```

Comment:
Rename: `without tasks`, reads better... `Server start without tasks should fail`